signature/signer.go

```diff
@@ -49,6 +57,38 @@ func NewLocalSigner(certs []*x509.Certificate, key crypto.PrivateKey) (LocalSign
 	}, nil
 }
 
+// keysMatch checks if the private key matches the provided public key.
+func keysMatch(privateKey crypto.PrivateKey, publicKey interface{}) bool {
+	privateKeySigner := privateKey.(crypto.Signer)
+
+	switch publicKey.(type) {
+	case *rsa.PublicKey:
+		switch privateKey.(type) {
+		case *rsa.PublicKey:
+			key, ok := privateKeySigner.Public().(*rsa.PrivateKey)
+			if !ok {
+				return false
+			}
+			return key.Equal(publicKey.(*rsa.PublicKey))
```

Comment:
An alternative way of checking the public key matches the private key is: [tls](https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/crypto/tls/tls.go;l=310)

Reply:
good call, updated.